modules/harvest/src/Entity/HarvestRunRepository.php

```diff
@@ -24,7 +24,7 @@ class HarvestRunRepository {
    *
    * @var \Drupal\Core\Entity\EntityStorageInterface
    */
-  private EntityStorageInterface $runStorage;
+  public EntityStorageInterface $runStorage;
```

Comment:
I can see making this protected for some reason, but public so you can access it is a bad pattern. Inject it elsewhere.

Reply:
This pattern strikes me as odd though that HarvertRunRepository's concern is " Convenient queries for harvest_run entities." and yet it offers no way to load a single harvest run.  If the pattern of exposing ->runStorage is bad, should we add a public method to HarvestRunRepository that is a  public passthrough for  ->runStorage->load()?

Otherwise it is kind of like giving you a set of keys that has every key you'll need to get into things, except your own front door.

Reply:
`$runstorage` is a service. If you need it somewhere else, inject it there. The reason is that if `HarvestRunRepository` promises to provide the service then it will always have that dependency, and will always have to have that public property, regardless of whether it needs it internally. Further down the line, if we deprecate more of the repository class, we're left providing a service that really should just be an injected service for those depending on it.

So, you're using the storage service in `HarvestPlanListBuilder`, where you could inject it in order to load a run record. Alternately, you could just say `HarvestRun::load($id)` (or `::loadMultiple()`), but the injected storage service is probably more of a best practice so the entity class doesn't have to look up its own storage every time.

Reply:
Good explanation.  This has been resolved.